integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanCapitalizedIncomeTest.java

```diff
@@ -96,4 +100,76 @@ public void testLoanCapitalizedIncomeAmortization() {
             );
         });
     }
+
+    @Test
+    public void testOverpaymentAmountWhenCapitalizedIncomeTransactionsAreReversed() {
+        final AtomicReference<Long> loanIdRef = new AtomicReference<>();
+        final PostClientsResponse client = clientHelper.createClient(ClientHelper.defaultClientCreationRequest());
+        runAt("01 March 2023", () -> {
+            final PostLoanProductsResponse loanProductsResponse = loanProductHelper
+                    .createLoanProduct(create4IProgressiveWithCapitalizedIncome());
+            PostLoansResponse postLoansResponse = loanTransactionHelper.applyLoan(applyLP2ProgressiveLoanRequest(client.getClientId(),
+                    loanProductsResponse.getResourceId(), "01 March 2023", 10000.00, 12.00, 4, null));
+            Long loanId = postLoansResponse.getLoanId();
+            loanIdRef.set(loanId);
+
+            loanTransactionHelper.approveLoan(loanId, approveLoanRequest(10000.00, "01 March 2023"));
+            disburseLoan(loanId, BigDecimal.valueOf(1000.00), "01 March 2023");
+
+            loanTransactionHelper.addCapitalizedIncome(loanId, "01 March 2023", 500.00);
+            loanTransactionHelper.addCapitalizedIncome(loanId, "01 March 2023", 500.00);
+            Long capitalizedIncomeId = loanTransactionHelper.getLoanDetails(loanId).getTransactions().get(2).getId();
```

Comment:
loanTransactionHelper.addCapitalizedIncome() returns the created transactions id in resourceId field

Reply:
For me it looked like it returns the loan Id itself, not the loan transaction Id, but I will check again after meetings.

Reply:
No, you are right. It is indeed the transaction id.